source/tutorials/nixos/nixos-configuration-on-vm.md

```diff
@@ -20,14 +20,22 @@ Virtual machines are a practical tool for debugging NixOS configurations.
 - A working [Nix installation](https://nixos.org/manual/nix/stable/installation/installation.html) on Linux, or [NixOS](https://nixos.org/manual/nixos/stable/index.html#sec-installation), with a graphical environment
 - Basic knowledge of the [Nix language](reading-nix-language)
 
-## Starting from the default NixOS configuration
+## Starting from a default NixOS configuration
 
-In this tutorial you will use the default configuration that is shipped with NixOS.[^nixosconf]
-[^nixosconf]: This [configuration template](https://github.com/NixOS/nixpkgs/blob/4e0525a8cdb370d31c1e1ba2641ad2a91fded57d/nixos/modules/installer/tools/tools.nix#L122-L226) is used.
+In this tutorial you will use a default configuration that is shipped with NixOS.
 
 :::{admonition} NixOS
 
 On NixOS, use the `nixos-generate-config` command to create a configuration file that contains some useful defaults and configuration suggestions.
+
+The result of this command depends on your curent Nixos configuration.
+We can configure the generation of the 'nixos-generate-config' in a repoducable way in a nix-shell. 
+We provide a configuration that is replicating nixos gnome graphical iso image.
+
+```shell-session
+nix-shell -p "with (import <nixpkgs> {}); (nixos { services.xserver.enable = true; services.xserver.desktopManager.gnome.enable = true;}).config.system.build.nixos-generate-config" -I nixpkgs=channel:nixos-21.05
+```
+
```

Comment:
This is a true abomination, which shows that Nix is awesome despite NixOS being in a very sad place. I love it. @olafklingt what a great piece of tinkering!

Suggestion to remove the veil of magic:

````suggestion

```bash
nix-shell -I nixpkgs=channel:nixos-21.05 -p "$(cat <<EOF
let
  pkgs = import <nixpkgs> { config = {}; overlays = []; };
  nixos = pkgs.nixos {
    services.xserver.enable = true;
    services.xserver.desktopManager.gnome.enable = true;
  };
in
nixos.config.system.build.nixos-generate-config
EOF
)" 
```

Or as a one-liner:

```shell-session
nix-shell -I nixpkgs=channel:nixos-21.05 -p "let pkgs = import <nixpkgs> { config = {}; overlays = []; }; nixos = pkgs.nixos { services.xserver.enable = true; services.xserver.desktopManager.gnome.enable = true; }; in nixos.config.system.build.nixos-generate-config"
```

````